pkg/config/config.go

```diff
@@ -1445,9 +1456,84 @@ func load(config Config, origin string, loadSecret bool) (*Warnings, error) {
 	// setTracemallocEnabled *must* be called before setNumWorkers
 	warnings.TraceMallocEnabledWithPy2 = setTracemallocEnabled(config)
 	setNumWorkers(config)
+	setupFipsEndpoints(config)
 	return &warnings, nil
 }
 
+// setupFipsEndpoints overwrites the Agent endpoint for outgoing data to be sent to the local FIPS proxy. The local FIPS
+// proxy will be in charge of forwarding data to the Datadog backend following FIPS standard. Starting from
+// fips.port_range_start we will assign a dedicated port per product (metrics, logs, traces, ...).
+func setupFipsEndpoints(config Config) {
+	// Each port is dedicated to a specific data type:
+	//
+	// port_range_start: HAProxy stats
+	// port_range_start + 1:  metrics
+	// port_range_start + 2:  traces
+	// port_range_start + 3:  profiles
+	// port_range_start + 4:  processes
+	// port_range_start + 5:  logs
+	// port_range_start + 6:  databases monitoring metrics
+	// port_range_start + 7:  databases monitoring samples
+	// port_range_start + 8:  network devices metadata
+	// port_range_start + 9:  network devices snmp traps (unused)
+	// port_range_start + 10: instrumentation telemetry
+	// port_range_start + 11: appsec events (unused)
+
+	if !config.GetBool("fips.enabled") {
+		log.Debug("fips mode is disabled")
+		return
+	}
+
+	const (
+		proxyStats = iota
+		metrics
+		traces
+		profiles
+		processes
+		logs
+		databasesMonitoringMetrics
+		databasesMonitoringSamples
+		networkDevicesMetadata
+		networkDevicesSnmpTraps
+		instrumentationTelemetry
+		appsecEvents
+	)
+
+	localAddress := config.GetString("fips.local_address")
+	portRangeStart := config.GetInt("fips.port_range_start")
+	urlFor := func(port int) string { return fmt.Sprintf("%s:%d", localAddress, portRangeStart+port) }
+
+	log.Infof("Fips mode is enabled: all communication to DataDog Backend will be routed to the local FIPS proxy on '%s' starting from port %d", localAddress, portRangeStart)
+
+	// Disabling proxy to make sure all data goes directly to the FIPS proxy
+	os.Unsetenv("HTTP_PROXY")
+	os.Unsetenv("HTTPS_PROXY")
+
+	// Metrics
+	config.Set("dd_url", "http://"+urlFor(metrics))
```

Comment:
Maybe we should comment here that these overrides should be synced with the config_template.yaml to make sure they're in sync?